File: _1503_last_moment_before_all_ants_fall_out_of_a_plank/main.py
class Solution:
    def getLastMoment(self, n: int, left: List[int], right: List[int]) -> int:
        if not left:
            return n - min(right)
        
        if not right:
            return max(left)

        return max(max(left), n - min(right))

        # Solution 1
        # return max(max(left) if len(left) > 0 else 0, n - min(right) if len(right) > 0 else 0)

File: _1503_last_moment_before_all_ants_fall_out_of_a_plank/main.cpp
class Solution {
public:
    int getLastMoment(int n, vector<int>& left, vector<int>& right) {
        // Solution 2
        int time = 0;
        for (int i : left)
        {
            time = max(time, i);
        }

        for (int i : right)
        {
            time = max(time, n - i);
        }

        return time;

        // Solution 1
        // int time = -1;
        // while (left.size() != 0 || right.size() != 0)
        // {
        //     time++;

        //     // --->
        //     for (auto it = right.begin(); it != right.end();)
        //     {
        //         if (*it == n)
        //         {
        //             it = right.erase(it);
        //         }
        //         else
        //         {
        //             (*it)++;
        //             ++it;
        //         }
        //     }

        //     // <---
        //     for (auto it = left.begin(); it != left.end();)
        //     {
        //         if (*it == 0)
        //         {
        //             it = left.erase(it);
        //         }
        //         else
        //         {
        //             (*it)--;
        //             ++it;
        //         }
        //     }
        // }

        // return time;
    }
};